openml-python-common/src/main/java/com/feedzai/openml/python/xml/parser/XMLParser.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2018 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.feedzai.openml.python.xml.parser;
+
+import com.google.common.collect.ImmutableSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class responsible for parsing a XML file in order to retrieve the modules of Python to be shared across
+ * sub-interpreters.
+ *
+ * @author Paulo Pereira (paulo.pereira@feedzai.com)
+ * @since 0.1.5
+ */
+public class XMLParser {
+
+    /**
+     * Logger.
+     */
+    private static final Logger logger = LoggerFactory.getLogger(XMLParser.class);
+
+    /**
+     * A XML file with the modules of Python to be shared across sub-interpreters.
+     */
+    private final String xmlFile;
+
+    /**
+     * Constructor.
+     */
+    public XMLParser() {
+        this.xmlFile = "python-packages.xml";
+    }
+
+    /**
+     * Gets the {@link InputStream} of {@link #xmlFile} that exists in the current classpath.
+     *
+     * @return The {@link InputStream} of {@link #xmlFile}.
+     */
+    private InputStream getXMLInputStream() {
+        final ClassLoader classLoader = getClass().getClassLoader();
+        return classLoader.getResourceAsStream(this.xmlFile);
+    }
+
+    /**
+     * Parses the {@link #xmlFile} to retrieve the modules of Python to be shared across sub-interpreters.
+     *
+     * @return A {@link Set} with the modules of Python to be shared across sub-interpreters.
+     * @throws Exception If there is an error while parsing {@link #xmlFile}.
+     */
+    private Set<String> parseXMLFile() throws Exception {
+        final HashSet<String> sharedModules = new HashSet<>();
```

Comment:
Do you really need sharedModules to be an HashSet, or any Set will do? I suspect any Set would do: in that case, declare the variable as Set<String> pls

Reply:
furthermore, I'm not very confortable in returning a mutable collection. What do you think of using an ImmutableSet.Builder ?